src/docfx/build/Build.cs

```diff
@@ -136,6 +138,25 @@ private static Task<DependencyMap> BuildAsset(Context context, Document file)
             return Task.FromResult(DependencyMap.Empty);
         }
 
+        private static void BuildRedirections(Docset docset, Context context)
+        {
+            Parallel.ForEach(docset.Redirections, redirection =>
```

Comment:
>Redirections [](start = 36, length = 12)

If two files are redirected to the same url, WriteJson may throw

Reply:
The result of GetSitePath(GetSiteUrl) could be the same for two different keys.

What if the key is TOC.md? We should forbid redirecting files that are not "master content" and show warnings.

Suggest using `Document.TryCreate`, add then determine if it can be redirected using `Document.IsMasterContent` property.

---
In reply to: [195022965](https://github.com/dotnet/docfx/pull/2886#discussion_r195022965) [](ancestors = 195022965)

Reply:
It should be validated during configuration loading? 

---
In reply to: [195282089](https://github.com/dotnet/docfx/pull/2886#discussion_r195282089) [](ancestors = 195282089,195022965)

Reply:
we can't use the document.trycreate, the key doesn't exist in the file system 

---
In reply to: [195282740](https://github.com/dotnet/docfx/pull/2886#discussion_r195282740) [](ancestors = 195282740,195282089,195022965)

Reply:
Let's do it here because the logic is heavy. We only do basic verification on config load.

---
In reply to: [195282740](https://github.com/dotnet/docfx/pull/2886#discussion_r195282740) [](ancestors = 195282740,195282089,195022965)

Reply:
Also, does routing affects the output path?

---
In reply to: [195282896](https://github.com/dotnet/docfx/pull/2886#discussion_r195282896) [](ancestors = 195282896,195282740,195282089,195022965)